cpp/src/reductions/reduction_operators.cuh

```diff
@@ -16,204 +16,256 @@
+#include <utilities/cudf_utils.h>  //for CUDA_HOST_DEVICE_CALLABLE
 #include <cudf/detail/iterator.cuh>
 #include <iterator/transform_unary_functions.cuh>
 #include <utilities/device_operators.cuh>
 
-#include <cmath>
 #include <thrust/functional.h>
+#include <cmath>
 
 namespace cudf {
 namespace experimental {
 namespace reduction {
 
 // intermediate data structure to compute `var`, `std`
-template<typename ResultType>
-struct var_std
-{
-    ResultType value;                /// the value
-    ResultType value_squared;        /// the value of squared
-
-    CUDA_HOST_DEVICE_CALLABLE
-    var_std(ResultType _value=0, ResultType _value_squared=0)
-    : value(_value), value_squared(_value_squared)
-    {};
-
-    using this_t = var_std<ResultType>;
-
-    CUDA_HOST_DEVICE_CALLABLE
-    this_t operator+(this_t const &rhs) const
-    {
-        return this_t(
-            (this->value + rhs.value),
-            (this->value_squared + rhs.value_squared)
-        );
-    };
+template <typename ResultType>
+struct var_std {
+  ResultType value;          /// the value
+  ResultType value_squared;  /// the value of squared
+
+  CUDA_HOST_DEVICE_CALLABLE
+  var_std(ResultType _value = 0, ResultType _value_squared = 0)
+      : value(_value), value_squared(_value_squared){};
+
+  using this_t = var_std<ResultType>;
+
+  CUDA_HOST_DEVICE_CALLABLE
+  this_t operator+(this_t const& rhs) const {
+    return this_t((this->value + rhs.value),
+                  (this->value_squared + rhs.value_squared));
+  };
 };
 
 // transformer for `struct var_std` in order to compute `var`, `std`
-template<typename ResultType>
-struct transformer_var_std
-{
-    using OutputType = var_std<ResultType>;
-
-    CUDA_HOST_DEVICE_CALLABLE
-    OutputType operator() (ResultType const & value)
-    {
-        return OutputType(value, value*value);
-    };
+template <typename ResultType>
+struct transformer_var_std {
+  using OutputType = var_std<ResultType>;
+
+  CUDA_HOST_DEVICE_CALLABLE
+  OutputType operator()(ResultType const& value) {
+    return OutputType(value, value * value);
+  };
 };
 
 // ------------------------------------------------------------------------
 // Definitions of device struct for reduction operation
 // all `op::xxx` must have `Op` and `transformer`
 // `Op`  is used to compute the reduction at device
-// `transformer` is used to convert elements for computing the reduction at device
-// By default `transformer` is static type conversion to ResultType
+// `transformer` is used to convert elements for computing the reduction at device.
+// By default `transformer` is static type conversion to ResultType.
 // In some cases, it could be square or abs or complex operations
-
 namespace op {
 
-// `sum`, `product`, `sum_of_squares`, `min`, `max`
-// are used at simple_reduction
+/**
+ * @brief  Simple reduction operator CRTP Base class
+ *
+ * @tparam Derived operator with SimpleOp interface
+ */
+template <typename Derived>
+struct SimpleOp {
```

Comment:
In libcudf, CamelCase is only used in template parameters and test fixture names.  Name this `simple_operator` or `simple_op`.